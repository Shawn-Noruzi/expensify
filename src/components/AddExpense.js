import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";

//to avoid inline functions we're switching to class component

export class AddExpensePage extends React.Component {
  onSubmit = expense => {
    this.props.addExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <h1> Add Expense </h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addExpense: expense => dispatch(addExpense(expense))
});
export default connect(
  undefined,
  mapDispatchToProps
)(AddExpensePage);

