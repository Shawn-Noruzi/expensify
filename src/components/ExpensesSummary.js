import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

//expenseCount - how many visisble expenses
//expensesTotal - total of visible expenses?
//2 snapshot tests
//sfc used here because we dont want this component to be able to
//change the state.

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format("$0,0.00");
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
