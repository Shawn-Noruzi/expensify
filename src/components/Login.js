import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

//doesnt need access to state -> dont need to provide stateToProps
//needs dispatchToProps -> bc we want to dispatch an action (startLogin)

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <button className="button" onClick={startLogin}><span>Login</span></button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
