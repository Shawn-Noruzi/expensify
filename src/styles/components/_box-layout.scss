.box-layout {
  //set a BG image in /public/images that covers the whole page
  background: url("/images/bg.jpg");
  background-size: cover;

  //positioning of elements in the center of the page
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  //setting the size of the entire div -> 100% of the viewport height and width
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out($color: rgb(255, 255, 255), $amount: 0.5);
  border-radius: 4px;
  text-align: center;
  width: 30rem;
  margin: $m-size;
  padding: $m-size;
}

.box-layout__title {
  margin: $s-size 0 0 0 ;
  line-height: 1;
  padding: $m-size;
  color: white;
}

.box-layout__paragraph {
  margin: $s-size 0 0 0 ;
  line-height: 1;
  padding: $s-size;
  font-size: 2 rem;
  color: white;
  text-align: start;
}

