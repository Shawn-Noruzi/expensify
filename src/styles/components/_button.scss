.button {
    background: rgb(22, 145, 182);
    color: white;
    border-radius: 4px;
    border: none;
    text-align: center;
    padding: 20px;
    transition: all 0.5s;
    cursor: pointer;
    margin-top: $s-size;
    margin-bottom: $l-size;
    font-size: $m-size;
  }
  .button span {
    cursor: pointer;
    display: inline-block;
    position: relative;
    transition: 0.5s;
  }
  
  .button span:after {
    content: "\00bb";
    position: absolute;
    opacity: 0;
    top: 0;
    right: -20px;
    transition: 0.5s;
  }
  
  .button:hover span {
    padding-right: 25px;
  }
  
  .button:hover span:after {
    opacity: 1;
    right: 0;
  }
  