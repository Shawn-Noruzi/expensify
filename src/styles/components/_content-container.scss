.content-container{
    margin: 0 1.8rem !important;
    padding: 0 $m-size;
    max-width:80rem;
}