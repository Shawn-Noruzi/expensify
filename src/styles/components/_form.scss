.form {
    display: flex;
    flex-direction: column;
    //selection for children of a tag ie.items under form
    > * {
        margin-bottom: $m-size;
    }
}



.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}