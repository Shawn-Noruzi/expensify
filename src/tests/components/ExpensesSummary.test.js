import React from "react";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import { shallow } from "enzyme";

test("should correctly render Expensessummary with 1 expense ", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={235} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render Expensessummary with multiple expense ", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={23} expensesTotal={23510000} />
  );
  expect(wrapper).toMatchSnapshot();
});
