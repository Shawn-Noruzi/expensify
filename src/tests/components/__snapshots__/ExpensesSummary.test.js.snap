// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expensessummary with 1 expense  1`] = `
<div>
   
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render Expensessummary with multiple expense  1`] = `
<div>
   
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $235,100.00
  </h1>
</div>
`;
